lightly/models/utils.py

```diff
@@ -471,20 +471,18 @@ def schedule_momentum(
 
     """
     Add sinusoidal noise to momentum value based on iteration number. Common
-    training tecnique in SSL
+    training technique in SSL.
 
     Args:
         iteration:
-            Size of the token batch for which to generate masks.
-            Should be (batch_size, sequence_length).
+            Current epoch number.
         max_iter:
-            Percentage of tokens to mask.
+            Total number of epochs.
         m:
-            If False the class token is never masked. If True the class token
-            might be masked.
+            Starting momentum value.
```

Comment:
I would propose to make this method more general and rename it to `cosine_schedule` with the following parameters:
```
step (instead of iteration)
max_steps (instead of max_iter)
start_value (instead of m)
end_value (new)
```
The method would then interpolate between `start_value` and `end_value` with a cosine decay.

Reply:
What would be the purpose of the end_value? I mean, if you put max steps, it is already an "end value" in the sense that you compute the end value with the step=max_steps

Reply:
Let's say you want cosine decay from 0.5 to 0.1 over 100 epochs. Then you would set:
```
max_steps = 100
start_value = 0.5
end_value = 0.1
```